core/lib/zksync_core/src/lib.rs

```diff
@@ -823,6 +849,27 @@ async fn run_tree(
     Ok(())
 }
 
+async fn add_basic_witness_input_producer_to_task_futures(
```

Comment:
nit: Witness generator is not spawned when `ZKSYNC_LOCAL_SETUP=true` is passed to env. I think it makes sense to have the same behavior for Producer.